pubsublite/producer.go

```diff
@@ -38,15 +38,21 @@ type Encoder interface {
 	Encode(model.APMEvent) ([]byte, error)
 }
 
+// RouterFunc returns the topic to use for a given APMEvent.
+type RouterFunc func(model.APMEvent) Topic
+
 // ProducerConfig for the PubSub Lite producer.
 type ProducerConfig struct {
-	// Topic is the PubSub Lite topic.
-	Topic Topic
+	// Topics are the PubSub Lite topics where the messages will be produced.
+	// The routing is determined by the EventRouter.
+	Topics []Topic
 	// Encoder holds an encoding.Encoder for encoding events.
 	Encoder Encoder
 	// Logger for the producer.
-	Logger     *zap.Logger
-	ClientOpts []option.ClientOption
+	Logger *zap.Logger
+	// EventRouter returns the topic where an event should be produced.
```

Comment:
```suggestion
	// EventRouter returns the topic where an event should be produced.
	//
	// If EventRouter returns a value that does not exist in Topics, then
	// ProcessBatch will return an error.
```

Alternatively we could lazily create producers?  This is fine for now though.

Reply:
yeah I suppose we could do it lazily, maybe we can create an issue to improve this? it may require some unlocking and re-locking I'm not to keen on for now.